src/sciline/pipeline.py

```diff
@@ -719,6 +736,23 @@ def get(
                 graph.update(self.build(t, handler=handler))
         else:
             graph = self.build(keys, handler=handler)  # type: ignore[arg-type]
+
+        if check:
+            all_providers = chain(
+                (
+                    (p[c], v)
+                    for p in self._subproviders
+                    for c, v in self._subproviders[p].items()
+                ),
+                ((p, v) for p, v in self._providers.items()),
+            )
+
+            for p, v in all_providers:
+                if p not in graph and _kind_of_provider(v) == 'parameter':
+                    raise RuntimeError(
+                        'Graph was expected to use all available parameters.'
+                    )
```

Comment:
If we take a step back, I am not sure this is sufficiently capturing the problem. In general, I think what we want to do is check whether a graph is connected, I suppose? Raising errors about individual disconnected pieces may not be meaningful?

Instead of the `check` flag and handling things via exceptions, would it be better to return some list (or other data structure) of disconnected pieces of the graph?